app/controllers/curation_concerns/generic_works_controller.rb

```diff
@@ -27,6 +28,39 @@ def identifiers
     end
   end
 
+  def check_date_coverage
+    yyyy1 = params[:date_coverage_1_year]
+    mm1 = params[:date_coverage_1_month]
+    dd1 = params[:date_coverage_1_day]
+   yyyy2 = params[:date_coverage_2_year]
+    mm2 = params[:date_coverage_2_month]
+    dd2 = params[:date_coverage_2_day]
+    if (yyyy1.length>0)
+      dateStr1 = yyyy1 
+    end  
+    if (!mm1.eql?("--"))
+     dateStr1 = dateStr1 + "/" + mm1
+       if (!dd1.eql?("--"))
+         dateStr1 = dateStr1 + "/" + dd1
+       end
+    end
+    if (yyyy2.length>0)
+      dateStr2 = yyyy2 
+    end
+    if (!mm2.eql? "--")
+      dateStr2 = dateStr2 + '/' + mm2
+        if (!dd2.eql? "--")
+         dateStr2 = dateStr2 + '/' + dd2
+      end
+    end
+    if (!dateStr1.nil?)
+     params['generic_work']['date_coverage'] = Date.parse(dateStr1).iso8601 
+    end
+    if (!dateStr2.nil?)
+      params['generic_work']['date_coverage'] = params['generic_work']['date_coverage'] + "TO" +  Date.parse(dateStr2).iso8601
+    end
```

Comment:
Doing this serialization ourselves means the de-serialization will have to be done ourselves as well.